app/code/Magento/InventoryApi/Api/Data/ReservationInterface.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\InventoryApi\Api\Data\ReservationExtensionInterface;
+
+/**
+ * The entity responsible for reservations, created to keep inventory amount (product quantity) up-to-date.
+ * It is created to have a state between order creation and inventory deduction (deduction of specific SourceItems)
+ *
+ * @api
+ */
+interface ReservationInterface extends ExtensibleDataInterface
+{
+    /**
+     * Constants for keys of data array. Identical to the name of the getter in snake case
+     */
+    const RESERVATION_ID = 'reservation_id';
+    const STOCK_ID = 'stock_id';
+    const SKU = 'sku';
+    const QUANTITY = 'quantity';
+    const STATUS = 'status';
+
+    /**#@+
+     * Reservation possible statuses. Maybe make sense to introduce extension point for Reservation Open-Close statuses
+     */
+    const STATUS_ORDER_CREATED = 1; // For Order Placed
+    const STATUS_RETURN_CREATED = 2; // For RMA Placed
+
+    const STATUS_OREDER_COMPLETE = 101; // For Order Complete
+    const STATUS_OREDER_CANCELED = 102; // For Order Canceled
+    const STATUS_RMA_COMPLETE = 103; // For RMA Canceled
+    /**#@-*/
+
+    /**
+     * Get Reservation id
+     *
+     * @return int|null
+     */
+    public function getReservationId();
+
+    /**
+     * Set Reservation Id
+     *
+     * @param int $reservationId
+     * @return void
+     */
+    public function setReservationId($reservationId);
+
+    /**
+     * Get stock id
+     *
+     * @param int $stockId
+     * @return void
+     */
+    public function getStockId();
+
+    /**
+     * Set stock id
+     *
+     * @param int $stockId
+     * @return void
+     */
+    public function setStockId($stockId);
+
+
+    /**
+     * Get Product SKU
+     *
+     * @return string|null
+     */
+    public function getSku();
+
+    /**
+     * Set product SKU
+     *
+     * @param string $sku
+     * @return void
+     */
+    public function setSku($sku);
+
+    /**
+     * Get Product Qty
+     *
+     * @return float|null
```

Comment:
@aleron75 
Correct, in the document I described it as qty is always > 0 and it is subtracted or added depending on Reservation status.

But thinking about an efficient way how can we get Sum of Grouped Reservation, made me think that it would be more performance efficient to have both negative and positive values for Reservations.

Because from the calculation point of view for us it would be easier to have both negative (<0) and positive (>0) Qty values.

Like when we placed an order -> we created Reservation with Qty **-30**,
when we processed the Order and deducted SourceItems -> we created a reservation with Qty **+30**

For example, executing this query

```sql
select 
   SUM(r.qty) as total_reservation_qty
from 
   Reservations as r
where 
  stockId = {%id%} and sku  = {%sku%}
```
pretty simple query, and we've given total reservation qty.

if we would have always Qty > 0, we need more sophisticated logic i.e.
- process aggregation query for Open Reservation
- process aggregation query for Closed Reservation
- subtract 1. from 2.